### DISCUSSION TITLE
Lesson 3 (Overrides and Virtual KeyWord)

### DISCUSSION THREAD

**Discussion by AbigirlMuchineripi** [Cyfrin/foundry-full-course-cu#1591] [Q&A]
Hello, may you kindly assist. I have successfully deployed the AddFiveStorage.sol contract but when I try to store any number and retrieve it , I get 0. It seems as if the numbers (2+5) are not adding up. Below is my screenshot. Thank you in advance.

![virtual keyword error](https://github.com/Cyfrin/foundry-full-course-f23/assets/94177156/3ac5fd2f-b5d1-4572-b20f-b784a992dc61)

**Comment by EngrPips** [comment]
Hello @AbigirlMuchineripi , please show your `SimpleStorage` contract and also your `AddFiveStorage` contract. I will be looking forward to hear back from you.

**Reply by AbigirlMuchineripi** [reply]
`
//SimpleStorage Contract

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18; // solidity versions

contract SimpleStorage {
    //favoriteNumber is initialise to 0 if no number is given
 uint256 public myFavoriteNumber;// 0

 //uint256[] OfFavoriteNumbers;
 struct Person{
    uint256 favoriteNumber;
    string name;

 }
 //dynamic array has empty parameters
 // static array has a parameter
 Person[] public listOfPeople;//[]

 //mapping is like a dictionary with all keys
 mapping(string => uint256) public nameToFavoriteNumber;

 function store(uint256 _favoriteNumber) public virtual{
   myFavoriteNumber = _favoriteNumber;
   
 }
 // view function disallows updating state, pure
 function retrieve() public view returns(uint256){
    return myFavoriteNumber;
 }
 //calldata, memory, storage,stack,code,logs
 function addPerson(string memory _name, uint256 _favoriteNumber) public{
     listOfPeople.push(Person(_favoriteNumber, _name));
     nameToFavoriteNumber[_name] = _favoriteNumber;
 }

}

contract SimpleStorage2 {}
contract SimpleStorage3 {}
contract SimpleStorage4 {}
contract SimpleStorage5 {}

`

`
//AddFiveStorage contract

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

//inheriting simpleStorage.sol
import {SimpleStorage} from "./simpleStorage.sol";

contract AddFiveStorage is SimpleStorage{
    //+5
    //Overrides
    // virtual overrides
    function store(uint256 _newNumber) public override{
        myFavoriteNumber = _newNumber + 5;

    }

}
`

**Reply by EngrPips** [reply]
```javascript
//SimpleStorage Contract

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18; // solidity versions

contract SimpleStorage {
//favoriteNumber is initialise to 0 if no number is given
uint256 public myFavoriteNumber;// 0

//uint256[] OfFavoriteNumbers;
struct Person{
uint256 favoriteNumber;
string name;

}
//dynamic array has empty parameters
// static array has a parameter
Person[] public listOfPeople;//[]

//mapping is like a dictionary with all keys
mapping(string => uint256) public nameToFavoriteNumber;

function store(uint256 _favoriteNumber) public virtual{
myFavoriteNumber = _favoriteNumber;

}
// view function disallows updating state, pure
function retrieve() public view returns(uint256){
return myFavoriteNumber;
}
//calldata, memory, storage,stack,code,logs
function addPerson(string memory _name, uint256 _favoriteNumber) public{
listOfPeople.push(Person(_favoriteNumber, _name));
nameToFavoriteNumber[_name] = _favoriteNumber;
}

}

contract SimpleStorage2 {}
contract SimpleStorage3 {}
contract SimpleStorage4 {}
contract SimpleStorage5 {}
```

```javascript
/AddFiveStorage contract

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

//inheriting simpleStorage.sol
import {SimpleStorage} from "./simpleStorage.sol";

contract AddFiveStorage is SimpleStorage{
//+5
//Overrides
// virtual overrides
function store(uint256 _newNumber) public override{
myFavoriteNumber = _newNumber + 5;
}
}
```

**Comment by EngrPips** [comment]
Hello @AbigirlMuchineripi , the contract worked fine for me, the exact code you sent. I am not sure why it isn't working for you.

<img width="1440" alt="Screenshot 2024-04-11 at 6 42 30 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/c6cd5858-9eed-4bc3-9fbf-fd39eb17e88f">
<img width="1440" alt="Screenshot 2024-04-11 at 6 42 36 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/3e114331-da61-4833-879d-9022a1037a0a">

You can see from the second image that i stored 77 as my favorite number and then when i went ahead to retrieve the number i got 82 which `77 + 5 = 82` .
